lib-extra/src/main/java/com/diffplug/spotless/extra/integration/DiffMessageFormatter.java

```diff
@@ -234,6 +234,10 @@ private void addIntendedLine(String indent, String line) {
 	 * sequence (\n, \r, \r\n).
 	 */
 	private String diff(File file) throws IOException {
+		return diff(formatter, file);
+	}
+
+	public static String diff(CleanProvider formatter, File file) throws IOException {
```

Comment:
Can we keep CleanProviderFormatter as private and just pass Formatter into the diff method?

Reply:
Done now in https://github.com/diffplug/spotless/pull/1962/commits/bdecbad91f66464a84c4d1f326cdfc9402b58066